profiler/src/main/java/com/splunk/opentelemetry/profiler/allocation/exporter/PlainTextAllocationEventExporter.java

```diff
@@ -0,0 +1,135 @@
+import static com.splunk.opentelemetry.profiler.LogExporterBuilder.INSTRUMENTATION_LIBRARY_INFO;
+
+import com.splunk.opentelemetry.profiler.LogDataCommonAttributes;
+import com.splunk.opentelemetry.profiler.allocation.sampler.AllocationEventSampler;
+import com.splunk.opentelemetry.profiler.util.StackSerializer;
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.common.AttributesBuilder;
+import io.opentelemetry.api.trace.SpanContext;
+import io.opentelemetry.sdk.logs.LogProcessor;
+import io.opentelemetry.sdk.logs.data.LogDataBuilder;
+import io.opentelemetry.sdk.resources.Resource;
+import java.time.Instant;
+import jdk.jfr.consumer.RecordedEvent;
+import jdk.jfr.consumer.RecordedStackTrace;
+import jdk.jfr.consumer.RecordedThread;
+
+public class PlainTextAllocationEventExporter implements AllocationEventExporter {
+  private static final AttributeKey<Long> ALLOCATION_SIZE_KEY =
+      AttributeKey.longKey("memory.allocated");
+
+  private final StackSerializer stackSerializer;
+  private final LogProcessor logProcessor;
+  private final LogDataCommonAttributes commonAttributes;
+  private final Resource resource;
+
+  private PlainTextAllocationEventExporter(Builder builder) {
+    this.stackSerializer = builder.stackSerializer;
+    this.logProcessor = builder.logProcessor;
+    this.commonAttributes = builder.commonAttributes;
+    this.resource = builder.resource;
+  }
+
+  @Override
+  public void export(RecordedEvent event, AllocationEventSampler sampler, SpanContext spanContext) {
+    RecordedStackTrace stackTrace = event.getStackTrace();
+    if (stackTrace == null) {
+      return;
+    }
+    Instant time = event.getStartTime();
+    String body = buildBody(event, stackTrace);
+
+    AttributesBuilder builder =
+        commonAttributes
+            .builder(event.getEventType().getName())
+            .put(ALLOCATION_SIZE_KEY, event.getLong("allocationSize"));
+    if (sampler != null) {
+      sampler.addAttributes(builder);
+    }
+    Attributes attributes = builder.build();
+
+    LogDataBuilder logDataBuilder =
+        LogDataBuilder.create(resource, INSTRUMENTATION_LIBRARY_INFO)
+            .setName(PROFILING_SOURCE)
+            .setEpoch(time)
+            .setBody(body)
+            .setAttributes(attributes);
+
+    if (spanContext != null && spanContext.isValid()) {
+      logDataBuilder.setSpanContext(spanContext);
+    }
+
+    logProcessor.emit(logDataBuilder.build());
+  }
+
+  private String buildBody(RecordedEvent event, RecordedStackTrace stackTrace) {
+    String stack = stackSerializer.serialize(stackTrace);
+    RecordedThread thread = event.getThread();
+    String name = thread == null ? "unknown" : thread.getJavaName();
+    long id = thread == null ? 0 : thread.getJavaThreadId();
+    return "\""
+        + name
+        + "\""
+        + " #"
+        + id
+        + " prio=0 os_prio=0 cpu=0ms elapsed=0s tid=0 nid=0 unknown"
```

Comment:
While we're here, can we change `nid=0` to `nid=0x0`? The thread id parser on the backend kinda expects it to be a hex number and spams the logs with warnings.

Reply:
It would be better to remove it completely rather than use bogus values. https://github.com/signalfx/gdi-specification/commit/b09e176ca3771c3ef19fc9d23e8722fc77a3b6e9 allows it.

Reply:
Yes, just remove them.